core/src/main/java/org/elasticsearch/index/query/MultiMatchQueryBuilder.java

```diff
@@ -747,7 +748,11 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
 
         Query query = multiMatchQuery.parse(type, newFieldsBoosts, value, minimumShouldMatch);
         if (query == null) {
-            return null;
+            if (newFieldsBoosts.isEmpty()) {
```

Comment:
but don't we forward to match_query which calls zeroTermsQuery when the query is null and that one builds a match no docs? We shouldn't have too many null queries returned. I am even wondering if it only happens when no fields are resolved... not sure. I still think we should dig and try to improve this. There's always time to give up, but let's at least try?